apps/daimo-mobile/src/storage/account.ts

```diff
@@ -0,0 +1,308 @@
+import { LandlineAccount } from "@daimo/api/src/landline/connector";
+import {
+  ChainGasConstants,
+  CurrencyExchangeRate,
+  DaimoInviteCodeStatus,
+  DaimoLinkNoteV2,
+  DaimoRequestV2Status,
+  DisplayOpEvent,
+  EAccount,
+  KeyData,
+  KeyRotationOpEvent,
+  LinkedAccount,
+  ProposedSwap,
+  RecommendedExchange,
+  SuggestedAction,
+  assert,
+  now,
+} from "@daimo/common";
+import { DaimoChain, daimoPaymasterV2Address } from "@daimo/contract";
+import { Address, Hex, getAddress } from "viem";
+
+import { StoredV16Account } from "./storedAccount";
+import { migrateOldAccount } from "./storedAccountMigrations";
+import { StoredModel } from "./storedModel";
+import { env, getEnvMobile } from "../env";
+
+const appVariant = getEnvMobile().DAIMO_APP_VARIANT;
+
+/**
+ * Singleton account key.
+ * Will be a series if/when we support multiple accounts.
+ */
+export const defaultEnclaveKeyName =
+  appVariant === "dev" ? "daimo-dev-12" : "daimo-12";
+
+/**
+ * Device API key name: serves as poor man's device attestation.
+ * Fixed key created once and never deleted -- used as an alternate to
+ * device attestations.
+ */
+export const deviceAPIKeyName =
+  appVariant === "dev" ? "daimo-apikey-dev" : "daimo-apikey";
+
+/** Account data stored on device. */
+export type Account = {
+  /** Local device signing key name */
+  enclaveKeyName: string;
+  /** Local device signing DER pubkey */
+  enclavePubKey: Hex;
+  /** Daimo name, registered onchain */
+  name: string;
+  /** Contract wallet address */
+  address: Address;
+
+  /** Home chain where we hold our balance */
+  homeChainId: number;
+  /** Home ERC-20 token where we hold our balance */
+  homeCoinAddress: Address;
```

Comment:
for next account version: thoughts on having `homeCoinToken: ForeignToken` or should we just have some function that grabs the ForeignToken from the address?